src/modules/article/application/createArticle.spec.ts

```diff
@@ -1,6 +1,83 @@
-import { ArticleSlug } from '@modules/article/domain'
+import { rest } from 'msw'
+import { HttpError } from '@shared/http/HttpError'
+import { HttpService } from '@@/src/shared/http/HttpService'
+import { mockServer } from '@@/src/shared/mockService'
+import { ArticleService as makeArticleService } from '../infrastructure/ArticleService'
+import { ArticleServiceMock } from '../infrastructure/ArticleService.mock'
+import { mockEditingArticle } from '../domain'
+import { createArticle as makeGetRecentArticles } from './createArticle'
 
-test('just asserting imports work fine in Jest env', () => {
-  const slug: ArticleSlug = 'asd'
-  expect(slug).toStrictEqual('asd')
+const mockHttpService = new HttpService('irrelevant')
+const articleService = makeArticleService(mockHttpService)
+const useCase = makeGetRecentArticles({ articleService })
+
+const server = mockServer(...ArticleServiceMock)
+
+const useCaseCallbacks = {
+  respondWithSuccess: jest.fn(),
+  respondWithClientError: jest.fn(),
+  respondWithServerError: jest.fn(),
+  respondWithGenericError: jest.fn(),
+}
+
+const editingArticle = mockEditingArticle()
+
+beforeAll(() => {
+  jest.useFakeTimers('modern')
+  jest.setSystemTime(new Date().getTime())
+})
+
+afterAll(() => {
+  jest.useRealTimers()
+})
+
+test('createArticle method is called', async () => {
+  const spiedMethod = jest.spyOn(articleService, 'createArticle')
```

Comment:
Is this some leftover?

Reply:
Nope, that's why it's the only test where we spy the method and we assert that is actually being called

Reply:
but where are we using the spy? Looks like an unused variable (other than calling `mockRestore`)?

Reply:
We need to spy the function in order to assert if this has been called

Reply:
Sorry, I'm not explaining myself properly. What I mean is that `spiedMethod` (the variable itself) is not being used. So we should either use:

```js
expect(spiedMethod).toHaveBeenCalledTimes(1)
```

below, or do not store the spied method anywhere:

```suggestion
  jest.spyOn(articleService, 'createArticle')
```